src/lib/containers/EntityBadgeIcons.tsx

```diff
@@ -0,0 +1,288 @@
+import React, { useState } from 'react'
+import { EntityBundle, EntityType } from '../utils/synapseTypes'
+import {
+  faAlignLeft,
+  faCheck,
+  faComment,
+  faGlobeAmericas,
+  faLock,
+  faTag,
+  faUnlink,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { isEmpty } from 'lodash-es'
+import sanitizeHtml from 'sanitize-html'
+import ReactTooltip from 'react-tooltip'
+import { SynapseClient } from '../utils'
+import { useSynapseContext } from '../utils/SynapseContext'
+import {
+  AUTHENTICATED_PRINCIPAL_ID,
+  PUBLIC_PRINCIPAL_ID,
+  ANONYMOUS_PRINCIPAL_ID,
+} from '../utils/SynapseConstants'
+import {
+  useGetSchemaBinding,
+  useGetValidationResults,
+} from '../utils/hooks/SynapseAPI/useSchema'
+import { useEffect } from 'react'
+import { EntityModal, EntityModalTabs } from './entity/metadata/EntityModal'
+import useGetEntityBundle from '../utils/hooks/SynapseAPI/useEntityBundle'
+
+const isPublic = (bundle: EntityBundle): boolean => {
+  return bundle.benefactorAcl!.resourceAccess.some(ra => {
+    return (
+      ra.principalId === AUTHENTICATED_PRINCIPAL_ID ||
+      ra.principalId === PUBLIC_PRINCIPAL_ID ||
+      ra.principalId === ANONYMOUS_PRINCIPAL_ID
+    )
+  })
+}
+
+type EntityBadgeIconsProps = {
+  entityId: string
+  flexWrap?: // possible settings for flex-wrap
+  | 'wrap'
+    | 'nowrap'
+    | '-moz-initial'
+    | 'inherit'
+    | 'initial'
+    | 'revert'
+    | 'unset'
+    | 'wrap-reverse'
+  justifyContent?: 'flex-start' | 'flex-end' | string
+  /** Shows an icon indicating if the entity is 'public' or 'private'. Default true  */
+  showIsPublicPrivate?: boolean
+  /** Shows an icon if the entity has sharing settings set on itself. Default true  */
+  showHasLocalSharingSettings?: boolean
+  /** Shows an icon if the entity has annotations, or if it has a validation schema (in experimental mode only). Default true  */
+  showHasAnnotations?: boolean
+  /** Shows an icon if the entity a wiki. Default true  */
+  showHasWiki?: boolean
+  /** Shows an icon if the entity has been mentioned in discussion threads. Default true  */
+  showHasDiscussionThread?: boolean
+  /* Shows an 'unlink' button if the entity is a link and the user has permission to delete it. Default true */
+  showUnlink?: boolean
```

Comment:
Before, we relied on the bundle contents to show the icon. Instead, fetch the full bundle (cached via react-query) and provide more granular control.